### PR TITLE
[GenericPoseGraph] Add proto defs for landmark cost functions.

### PR DESCRIPTION
**Pull request by pifon2a**
@mbokeloh Martin, please, take a look, maybe you'll come up with less idiotic naming than I did.

### CODE REVIEW

File: cartographer/pose_graph/proto/cost_function.proto [hunk 1]
```diff
@@ -49,19 +49,58 @@ message Acceleration3D {
...
+  Parameters parameters = 4;
+}
+
+message InterpolatedNode {
+  NodeId start;
+  NodeId end;
+  double factor;
+}
+
+message RelativePoseFromInterpolatedNode2D {
```
Comment (mbokeloh) [inline cartographer/pose_graph/proto/cost_function.proto]: maybe RelativePose2DInterpolated?

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 3.

**Comment by wally-the-cartographer** [comment]
Error: exit status 1

Log:
~~~
=> handle work item: PR 1312 on googlecartographer/cartographer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 3a9d1bc4 [GenericPoseGraph] Added rotation cost 3d. (#1313)
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
Already up to date.
Current branch master is up to date.
 => Checking out pifon2a/protodef as local branch pr_1312.
 $ git remote add pifon2a git@github.com:pifon2a/cartographer.git [data/cartographer]
fatal: remote pifon2a already exists.
 $ git fetch pifon2a [data/cartographer]
 $ git branch --track pr_1312 pifon2a/protodef [data/cartographer]
Branch 'pr_1312' set up to track remote branch 'protodef' from 'pifon2a'.
 => Switching to branch pr_1312.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 3a9d1bc4 [GenericPoseGraph] Added rotation cost 3d. (#1313)
 $ git clean -fd [data/cartographer]
 $ git checkout pr_1312 [data/cartographer]
Switched to branch 'pr_1312'
Your branch is up to date with 'pifon2a/protodef'.
 $ git pull --force [data/cartographer]
Already up to date.
Current branch pr_1312 is up to date.
 => merge master.
 $ git merge master [data/cartographer]
Auto-merging cartographer/pose_graph/proto/cost_function.proto
CONFLICT (content): Merge conflict in cartographer/pose_graph/proto/cost_function.proto
Automatic merge failed; fix conflicts and then commit the result.
 => Abandoning branch pr_1312.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at af129ba2 [GenericPoseGraph] Add proto defs for landmark cost functions.
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git branch -D pr_1312 [data/cartographer]
Deleted branch pr_1312 (was af129ba2).

~~~

**Pull request merged**